src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -293,6 +300,7 @@ void TieredHNSWIndex<DataType, DistType>::executeReadySwapJobs() {
 template <typename DataType, typename DistType>
 int TieredHNSWIndex<DataType, DistType>::deleteLabelFromHNSW(labelType label) {
     auto *hnsw_index = getHNSWIndex();
+    this->mainIndexGuard.lock_shared();
```

Comment:
why the addition?

Reply:
I moved the lock/unlock into `deleteLabelFromHNSW` (rather than locking before and after in the caller function)